db_upgrade_scripts/mosip_ida/sql/1.2.1.0_to_1.3.0_rollback.sql

```diff
@@ -1,28 +1,38 @@
 -- ------------------------------------------------------------------------------------------
--- Rollback script for reverting Spring Batch version 5.0 migration as part of Java 21 migration.
+-- Rollback script for Migrating Spring batch version back from 5.0 as part of Java 21 Migration.
 -- ------------------------------------------------------------------------------------------
 
--- Revert the changes to BATCH_STEP_EXECUTION
-ALTER TABLE BATCH_STEP_EXECUTION ALTER COLUMN START_TIME SET NOT NULL;
-ALTER TABLE BATCH_STEP_EXECUTION DROP COLUMN CREATE_TIME;
-
--- Revert the changes to BATCH_JOB_EXECUTION_PARAMS
-ALTER TABLE BATCH_JOB_EXECUTION_PARAMS ADD COLUMN DATE_VAL TIMESTAMP;
-ALTER TABLE BATCH_JOB_EXECUTION_PARAMS ADD COLUMN LONG_VAL BIGINT;
-ALTER TABLE BATCH_JOB_EXECUTION_PARAMS ADD COLUMN DOUBLE_VAL DOUBLE PRECISION;
-ALTER TABLE BATCH_JOB_EXECUTION_PARAMS ALTER COLUMN PARAMETER_TYPE TYPE CHAR(6);
-ALTER TABLE BATCH_JOB_EXECUTION_PARAMS RENAME PARAMETER_TYPE TO TYPE_CD;
-ALTER TABLE BATCH_JOB_EXECUTION_PARAMS ALTER COLUMN PARAMETER_NAME TYPE VARCHAR(100);
-ALTER TABLE BATCH_JOB_EXECUTION_PARAMS RENAME PARAMETER_NAME TO KEY_NAME;
-ALTER TABLE BATCH_JOB_EXECUTION_PARAMS ALTER COLUMN PARAMETER_VALUE TYPE VARCHAR(250);
-ALTER TABLE BATCH_JOB_EXECUTION_PARAMS RENAME PARAMETER_VALUE TO STRING_VAL;
-ALTER TABLE BATCH_JOB_EXECUTION ADD COLUMN JOB_CONFIGURATION_LOCATION VARCHAR(2500);
-
--- Drop the created sequences
-DROP SEQUENCE BATCH_STEP_EXECUTION_SEQ;
-DROP SEQUENCE BATCH_JOB_EXECUTION_SEQ;
-DROP SEQUENCE BATCH_JOB_SEQ;
-
--- Drop the created indexes
-DROP INDEX idx_job_name;
-DROP INDEX idx_job_key;
+TRUNCATE TABLE batch_job_execution CASCADE;
+TRUNCATE TABLE batch_job_execution_context CASCADE;
+TRUNCATE TABLE batch_job_execution_params CASCADE;
+TRUNCATE TABLE batch_job_instance CASCADE;
+TRUNCATE TABLE batch_step_execution CASCADE;
+TRUNCATE TABLE batch_step_execution_context CASCADE;
+
+GRANT usage, SELECT ON ALL SEQUENCES
+   IN SCHEMA ida
+   TO idauser;
+
+-- Revert ALTER on BATCH_STEP_EXECUTION
+ALTER TABLE BATCH_STEP_EXECUTION DROP COLUMN CREATE_TIME;  -- Remove the column added
+ALTER TABLE BATCH_STEP_EXECUTION ALTER COLUMN START_TIME SET NOT NULL;  -- Revert to NOT NULL
+
+-- Revert changes on BATCH_JOB_EXECUTION_PARAMS
+ALTER TABLE BATCH_JOB_EXECUTION_PARAMS ADD COLUMN DATE_VAL TIMESTAMP;  -- Add back the DATE_VAL column
+ALTER TABLE BATCH_JOB_EXECUTION_PARAMS ADD COLUMN LONG_VAL BIGINT;     -- Add back the LONG_VAL column
+ALTER TABLE BATCH_JOB_EXECUTION_PARAMS ADD COLUMN DOUBLE_VAL DOUBLE PRECISION;  -- Add back the DOUBLE_VAL column
+
+ALTER TABLE BATCH_JOB_EXECUTION_PARAMS RENAME PARAMETER_TYPE TO TYPE_CD;            -- Revert the column name
```

Comment:
We need to rollback the data type as well. If it is not required then remove the below statemen from the upgrade scripts as well:
ALTER TABLE BATCH_JOB_EXECUTION_PARAMS ALTER COLUMN TYPE_CD TYPE VARCHAR(100);

Reply:
corrected